### PR TITLE
ButtonToggle: New Component v1

### PR DESCRIPTION
**Pull request by eniomoura**
#### What changed?

New Component - ButtonToggle 

image

image

A11y:
Screenshot by Dropbox Capture

Screenshot by Dropbox Capture

#### Why?

ButtonToggle is only used for buttons that have a selected/unselected state. They should not be used as directional/navigation elements.

### Links

- [Jira](https://jira.pinadmin.com/browse/CPW-6876)
- [TDD](https://docs.google.com/document/d/1OWdn092wUck0ypeXt4MFSMPWYHyt4zxYiWpiKE2f17k/edit)
- [Figma](https://www.figma.com/design/vjhfBsOtHw0wVg67vqwz1v/branch/JTOaOCKWdZDH3arGpofIE5/Gestalt-for-Web?node-id=33641-2391&m=dev)

### Checklist

- [ Yes ] Added unit and Flow Tests
- [ Yes ] Added documentation + accessibility tests
- [ Yes ] Verified accessibility: keyboard & screen reader interaction
- [ Yes ] Checked dark mode, responsiveness, and right-to-left support
- [ Yes ] Checked stakeholder feedback (e.g. Gestalt designers, relevant feature teams)

### CODE REVIEW

File: packages/gestalt/src/ButtonToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  /**
+   * Callback invoked when the user clicks (press and release) on ButtonToggle with the mouse or keyboard.
+   */
+  onClick?: (arg1: {
+    event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLButtonElement>;
+  }) => void;
+  /**
+   * Toggles between selected/unselected.
+   */
+  selected?: boolean;
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:63]: this should be required no?

File: packages/gestalt/src/ButtonToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+        ) : null}
+        {text}
+      </Flex>
+      <NewTabAccessibilityLabel target={target} />
+    </Fragment>
+  );
+}
+
+/**
+ * [ButtonToggles](https://gestalt.pinterest.systems/web/buttontoggle) is a larger alternative to selection components such as [Checkbox](/android/checkbox), [RadioButton](/web/radiobutton), and [Switch](/android/switch). It enables users to choose between two states - selected or unselected.
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:109]: ButtonToggle

Comment (AlbertCarreras) [reply packages/gestalt/src/ButtonToggle.tsx:109]: we need full links /web/radiobutton cos its supposed to go to VSCode extension too

File: packages/gestalt/src/ButtonToggle.tsx [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+ *
+ * ![ButtonToggle light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonToggle.spec.mjs-snapshots/ButtonToggle-chromium-darwin.png)
+ * ![ButtonToggle dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonToggle-dark.spec.mjs-snapshots/ButtonToggle-dark-chromium-darwin.png)
+ *
+ */
+const ButtonWithForwardRef = forwardRef<HTMLButtonElement, Props>(function ButtonToggle(
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:115]: ButtonToggleWithForwardRef

File: packages/gestalt/src/ButtonToggle.css [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+.gray {
+  background-color: var(--color-background-button-secondary-default);
+}
+
+.gray:hover {
+  background-color: var(--color-background-button-secondary-hover);
+}
+
+.gray:active {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.css:101]: you need to cleanup this css file!!!!

File: packages/gestalt/src/ButtonToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  });
+
+  const parentButtonClasses = classnames(sharedTypeClasses, styles.parentButton);
+
+  const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);
+
+  const textColor =
+    (disabled && 'subtle') ||
+    (isDarkModeRed && 'default') ||
+    (selected && 'default') ||
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:182]: is the order correct? 

should it be disabled/selected/isDarkRedMode?

File: packages/gestalt/src/ButtonToggle.tsx [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+  md: 12,
+  lg: 12,
+} as const;
+
+type Target = null | 'self' | 'blank';
+
+type Props = {
+  /**
+   * Specifies the `id` of an associated element (or elements) whose contents or visibility are controlled by ButtonToggle so that screen reader users can identify the relationship between elements. See the [Accessibility guidelines](https://gestalt.pinterest.systems/web/buttontoggle#ARIA-attributes) for details on proper usage.
+   */
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:32]: can we add onBlur and onFocus?

File: docs/pages/web/buttontoggle.tsx [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+    Medium is used on more dense UI such as business surfaces or internal tools.
+3. \`sm\` (32px)
+    Small should be used sparingly and only in places where the UI is very dense.`}
+          title="Size"
+        >
+          {/* @ts-expect-error - TS2322 - Type '{ children: ({ size }: { [key: string]: any; }) => Element; size: string[]; }' is not assignable to type 'IntrinsicAttributes & Props'. */}
+          <CombinationNew size={['sm', 'md', 'lg']}>
+            {({ size }) => (
+              <ButtonToggle
+                accessibilityLabel={`Example size ${size}`}
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:173]: can we use the transparent variant here?

File: packages/gestalt/src/ButtonToggle.tsx (hunk 1 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx]: Yes, fixed!

File: packages/gestalt/src/ButtonToggle.tsx (hunk 2 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:109]: Fixed!

File: packages/gestalt/src/ButtonToggle.tsx (hunk 3 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:115]: Fixed!

File: packages/gestalt/src/ButtonToggle.tsx (hunk 5 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:182]: Fixed, it had ended up like this after some iteration.

File: docs/pages/visual-test/ButtonToggle.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { Box, ButtonToggle } from 'gestalt';
+
+export default function Snapshot() {
+  return (
+    <Box padding={1}>
+      <ButtonToggle color="transparent" iconStart="sparkle" text="Save" />
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply docs/pages/visual-test/ButtonToggle.tsx]: Done!

File: docs/pages/web/buttontoggle.tsx (hunk 7 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:173]: Done!

File: docs/pages/web/buttontoggle.tsx [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+                color="red"
+                size={size}
+                text="Save"
+              />
+            )}
+          </CombinationNew>
+        </MainSection.Subsection>
+        <MainSection.Subsection
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:182]: Done!

File: packages/gestalt/src/ButtonToggle.tsx [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+            accessibilityLabel=""
+            color={textColor}
+            icon={icon}
+            // @ts-expect-error - TS7053 - Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ readonly sm: 10; readonly md: 12; readonly lg: 12; }'.
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        ) : null}
+        {text}
+      </Flex>
+      <NewTabAccessibilityLabel target={target} />
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx]: Done!

File: packages/gestalt/src/ButtonToggle.css (hunk 4 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.css:101]: Done!

File: packages/gestalt/src/ButtonToggle.tsx (hunk 6 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:29]: Done!

File: docs/pages/web/buttontoggle.tsx [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+Our current disabled ButtonToggle implementation does fail to pass color contrast on accessibility integration tests. To exclude disabled buttontoggles from the integration tests we recomment conditionally setting a \`data-test-id={ isDisabled ? "disabled-button-<name>" : undefined }\` and excluding them from the integration test.
+
+On [cypress-axe](https://www.npmjs.com/package/cypress-axe) that can be achieved with <code>cy.a11yCheck({ exclude: [['[data-test-id="disabled-button-submit"]']] })<code>`}
+          title="Color contrast in disabled state"
+        />
+      </AccessibilitySection>
+
+      <LocalizationSection
+        name={generatedDocGen?.displayName}
+        noDefaultLabelProvider
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:154]: Done!

File: packages/gestalt/src/ButtonToggle.tsx [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+      }}
+      onMouseDown={handleMouseDown}
+      onMouseUp={handleMouseUp}
+      onTouchCancel={handleTouchCancel}
+      onTouchEnd={handleTouchEnd}
+      // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
+      onTouchMove={handleTouchMove}
+      // @ts-expect-error - TS2322 - Type '(arg1: TouchEvent<HTMLDivElement>) => void' is not assignable to type 'TouchEventHandler<HTMLButtonElement>'.
+      onTouchStart={handleTouchStart}
+      type="button"
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:206]: I have added the aria-pressed state missing on the Button based on the selected state. I also double-checked the TDD, some stuff would only apply to normal Buttons but made no sense in that context. 
Tested accessibility with a screen reader, it is now able to tell the selected state of the toggle, together with the text and accessibility label used.

**Comment by AlbertCarreras** [comment]
@eniomoura Can we get tests to pass?

**Comment by AlbertCarreras** [comment]
Can we add a Controlled component variant? Exactly like https://gestalt.pinterest.systems/web/datefield#Controlled-component

File: docs/pages/web/buttontoggle.tsx [hunk 13]
```diff
@@ -0,0 +1,321 @@
...
+      </MainSection>
+      <AccessibilitySection name={generatedDocGen?.displayName}>
+        <MainSection.Subsection
+          description={`
+When ButtonToggle text does not provide sufficient context about the ButtonToggle’s behavior, supply a short, descriptive label for screen-readers using \`accessibilityLabel\`.
+Texts like “Best“, “Views“, or “Save“ can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text with deeper context to replace the ButtonToggle text, like “Follow Ryan” or “Shop Wedding Invitations”.
+
+If ButtonToggle is used as a control button to show/hide a Popover-based component, we recommend passing the following ARIA attributes to assist screen readers:
+- \`accessibilityLabel\`: if present, read by screen readers read instead of the \`text\` prop.
+- \`accessibilityControls\`: informs the screen reader that ButtonToggle controls the display of an anchored Popover-based component. It populates [aria-controls](https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html).
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:135]: let's use this links

accessibilityLabel : https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
 accessibilityControls : https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

tweak the definition of accessibilityControls to the description in the link i gave you.

File: docs/examples/buttontoggle/colors.tsx [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { Box, ButtonToggle, Flex, Text } from 'gestalt';
+
+export default function Example() {
+  return (
+    <Box padding={4}>
+      <Flex gap={6} height="100%" width="100%" wrap>
+        [
```
Comment (AlbertCarreras) [inline docs/examples/buttontoggle/colors.tsx:7]: Screenshot by Dropbox Capture

File: docs/pages/web/buttontoggle.tsx [hunk 15]
```diff
@@ -0,0 +1,321 @@
...
+      <MainSection name="Best practices">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            description={`To make it clearer, you may want to change the label text to indicate that the ButtonToggle has been selected. For instance, changing "follow" to "Following."`}
+            sandpackExample={
+              <SandpackExample
+                code={changeLabelsDo}
+                hideEditor
+                name={`To make it clearer, you may want to change the label text to indicate that the ButtonToggle has been selected. For instance, changing "follow" to "Following."`}
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:71]: name is just too long, can we just say "content"

File: docs/pages/web/buttontoggle.tsx [hunk 16]
```diff
@@ -0,0 +1,321 @@
...
+            title="When not to use"
+            type="don't"
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best practices">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            description={`To make it clearer, you may want to change the label text to indicate that the ButtonToggle has been selected. For instance, changing "follow" to "Following."`}
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:66]: "follow" to "Following >> "**F**ollow" to "Following.

File: docs/pages/web/buttontoggle.tsx [hunk 17]
```diff
@@ -0,0 +1,321 @@
...
+
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            description="If the ButtonToggle(s) trigger a significant action or irreversible change, it is recommended to include a confirmation, such as a toast message."
+            sandpackExample={
+              <SandpackExample
+                code={confirmationDo}
+                hideEditor
+                name="If the ButtonToggle(s) trigger a significant action or irreversible change, it is recommended to include a confirmation, such as a toast message."
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:119]: can we replace names with a shorter name? In all examples.

File: docs/pages/web/buttontoggle.tsx [hunk 18]
```diff
@@ -0,0 +1,321 @@
...
+              />
+            }
+            type="do"
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            description="If the ButtonToggle(s) trigger a significant action or irreversible change, it is recommended to include a confirmation, such as a toast message."
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:114]: Can we check with Hector if we can mention ModalAlert and Toast? Also toast in uppercase. Cos we say toast but use a ModalAlert... not consistent text w example.

File: docs/pages/web/buttontoggle.tsx [hunk 19]
```diff
@@ -0,0 +1,321 @@
...
+            type="don't"
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <QualityChecklist component={generatedDocGen?.displayName} />
+
+      <MainSection name="Related">
+        <MainSection.Subsection
+          description={`
+**[Button](/web/button)**
```
Comment (AlbertCarreras) [inline docs/pages/web/buttontoggle.tsx:289]: this related section is a big copy paste. 


ButtonToggle is a larger alternative to selection components such as [Checkbox](https://deploy-preview-3577--gestalt.netlify.app/android/checkbox), [RadioButton](https://gestalt.pinterest.systems/web/radiobutton), and [Switch](https://deploy-preview-3577--gestalt.netlify.app/android/switch).

I'd replace all this ones with Checkbox,RadioButton, and Switch

File: docs/examples/buttontoggle/defaultStateExample.tsx [hunk 20]
```diff
@@ -0,0 +1,9 @@
+import { ButtonToggle, Flex } from 'gestalt';
+
+export default function Example() {
+  return (
+    <Flex alignItems="center" height="100%" justifyContent="center" width="100%">
```
Comment (AlbertCarreras) [inline docs/examples/buttontoggle/defaultStateExample.tsx:5]: add both variants in color

File: docs/examples/buttontoggle/disabledStateExample.tsx [hunk 21]
```diff
@@ -0,0 +1,9 @@
+import { ButtonToggle, Flex } from 'gestalt';
+
+export default function Example() {
+  return (
+    <Flex alignItems="center" height="100%" justifyContent="center" width="100%">
+      <ButtonToggle accessibilityLabel="Follow" disabled selected={false} size="lg" text="Follow" />{' '}
```
Comment (AlbertCarreras) [inline docs/examples/buttontoggle/disabledStateExample.tsx:6]: add both variants in color

File: docs/examples/buttontoggle/selectedStateExample.tsx [hunk 22]
```diff
@@ -0,0 +1,9 @@
+import { ButtonToggle, Flex } from 'gestalt';
+
+export default function Example() {
+  return (
+    <Flex alignItems="center" height="100%" justifyContent="center" width="100%">
+      <ButtonToggle accessibilityLabel="Following" selected size="lg" text="Following" />
```
Comment (AlbertCarreras) [inline docs/examples/buttontoggle/selectedStateExample.tsx:6]: add both variants in color

File: packages/gestalt/src/ButtonToggle.css [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+  composes: paddingX4 paddingY3 from "./boxWhitespace.css";
+}
+
+.enabled {
+  composes: pointer from "./Cursor.css";
+}
+
+/* STATES */
+
+.disabled {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.css:44]: disabled can be consolidated

.disabled { 
  composes: borderColorLightGrayDisabled from "./Borders.css";
  cursor: default;
  pointer-events: none;
}

.disabledTransparent { 
background-color: var(--color-background-button-tertiary-default);)
}

.disabledRed { 
 background-color: var(--color-background-button-disabled-default);
}

File: packages/gestalt/src/ButtonToggle.tsx [hunk 24]
```diff
@@ -0,0 +1,231 @@
...
+
+  const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);
+
+  const textColor =
+    (disabled && 'subtle') ||
+    (selected && 'default') ||
+    (isDarkModeRed && 'default') ||
+    DEFAULT_TEXT_COLORS[color];
+
+  const buttonText = (
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:161]: can you just inline?

we are doing text on text

 <Text>
    <Text
      align="center"
      color={textColor}
      overflow="normal"
      size={size === 'sm' ? '200' : '300'}
      weight="bold"
    >
      {text}
    </Text>
    </Text>

**Review by AlbertCarreras — APPROVED** [review]

File: packages/gestalt/dist/index.d.ts [hunk 25]
```diff
@@ -2668,6 +2684,11 @@ export const ButtonLink: ReactForwardRef<HTMLAnchorElement, ButtonLinkProps>;
  */
 export const ButtonGroup: React.FunctionComponent<ButtonGroupProps>;
 
+/**
+ * https://gestalt.pinterest.systems/web/button
+ */
+export const ButtonToggle: ReactForwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonToggleProps>;
```
Comment (AlbertCarreras) [inline packages/gestalt/dist/index.d.ts:2690]: HTMLAnchorElement no, remove

File: packages/gestalt/dist/index.d.ts [hunk 26]
```diff
@@ -933,6 +933,22 @@ interface ButtonProps {
   type?: 'button' | 'submit' | undefined;
 }
 
+interface ButtonToggleProps {
+  accessibilityControls?: string | undefined;
+  accessibilityLabel?: string | undefined;
+  color?:
+    | 'red'
+    | 'transparent'
+    | undefined;
```
Comment (AlbertCarreras) [inline packages/gestalt/dist/index.d.ts:942]: let's remove all the | undefined... they are incorrect indeed, everywhere...

File: packages/gestalt/dist/index.d.ts [hunk 27]
```diff
@@ -932,6 +932,22 @@ interface ButtonProps {
...
+  dataTestId?: string;
+  disabled?: boolean | undefined;
+  iconStart?: Icons | undefined;
+  onClick?: ButtonEventHandlerType | undefined;
```
Comment (AlbertCarreras) [inline packages/gestalt/dist/index.d.ts:945]: onBlur, onFocus missing

File: packages/gestalt/dist/index.d.ts (hunk 25 above)
Comment (eniomoura) [reply packages/gestalt/dist/index.d.ts]: done

File: packages/gestalt/dist/index.d.ts (hunk 26 above)
Comment (eniomoura) [reply packages/gestalt/dist/index.d.ts]: done

File: packages/gestalt/dist/index.d.ts (hunk 27 above)
Comment (eniomoura) [reply packages/gestalt/dist/index.d.ts]: done

File: docs/examples/buttontoggle/colors.tsx (hunk 14 above)
Comment (eniomoura) [reply docs/examples/buttontoggle/colors.tsx]: fixed!

File: packages/gestalt/src/ButtonToggle.tsx [hunk 28]
```diff
@@ -0,0 +1,231 @@
...
+  {
+    accessibilityLabel,
+    color = 'transparent',
+    dataTestId,
+    disabled = false,
+    iconStart,
+    onBlur,
+    onClick,
+    onFocus,
+    selected = false,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/ButtonToggle.tsx:98]: selected = false, there should be no default as it is required

File: docs/pages/web/buttontoggle.tsx (hunk 13 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:135]: DONE

File: docs/pages/web/buttontoggle.tsx (hunk 15 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:71]: DONE

File: docs/pages/web/buttontoggle.tsx (hunk 16 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:66]: DONE

File: docs/pages/web/buttontoggle.tsx (hunk 17 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx]: DONE

File: docs/pages/web/buttontoggle.tsx (hunk 19 above)
Comment (eniomoura) [reply docs/pages/web/buttontoggle.tsx:289]: DONE

File: packages/gestalt/src/ButtonToggle.css (hunk 23 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.css:44]: DONE

File: packages/gestalt/src/ButtonToggle.tsx (hunk 24 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:161]: DONE

File: packages/gestalt/src/ButtonToggle.tsx (hunk 28 above)
Comment (eniomoura) [reply packages/gestalt/src/ButtonToggle.tsx:98]: DONE

File: docs/examples/buttontoggle/defaultStateExample.tsx (hunk 20 above)
Comment (eniomoura) [reply docs/examples/buttontoggle/defaultStateExample.tsx:5]: DONE

File: docs/examples/buttontoggle/disabledStateExample.tsx (hunk 21 above)
Comment (eniomoura) [reply docs/examples/buttontoggle/disabledStateExample.tsx:6]: DONE

File: docs/examples/buttontoggle/selectedStateExample.tsx (hunk 22 above)
Comment (eniomoura) [reply docs/examples/buttontoggle/selectedStateExample.tsx:6]: DONE

File: docs/examples/buttontoggle/locationDo.tsx [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+        <Button
+          color="red"
+          onClick={() => {
+            setFilter(true);
+          }}
+          size="lg"
+          text="Filter"
+        />
+      </ButtonGroup>
+      <Text align="center" color="default">
```
[the remark this answers is not in the document]
Comment (eniomoura) [reply docs/examples/buttontoggle/locationDo.tsx:46]: DONE

**Pull request merged**